src/collections/map.rs

```diff
@@ -562,7 +712,650 @@ mod tests {
+
+        edit_map.entry(12).unwrap().or_insert(24).unwrap();
+        edit_map.entry(14).unwrap().or_insert(28).unwrap();
+
+        edit_map.flush().unwrap();
+
+        let mut read_map: Map<u32, u32> = Map::create(store.clone(), ()).unwrap();
+        read_map.entry(13).unwrap().or_insert(26).unwrap();
+
+        let mut actual: Vec<(u32, u32)> = Vec::with_capacity(3);
+
+        read_map
+            .range(12..=14)
+            .for_each(|(k, v)| actual.push((k, *v)));
+
+        let expected: Vec<(u32, u32)> = vec![(12, 24), (13, 26), (14, 28)];
+
+        assert_eq!(actual, expected);
+    }
+
+    #[test]
+    fn map_range_empty() {
+        let store = Store::new(MapStore::new());
+        let mut map: Map<u32, u32> = Map::create(store.clone(), ()).unwrap();
+
+        let mut actual: Vec<(u32, u32)> = Vec::with_capacity(3);
+
+        map.range(..).for_each(|(k, v)| actual.push((k, *v)));
+
+        let expected: Vec<(u32, u32)> = vec![];
+
+        assert_eq!(actual, expected);
+    }
+
+    #[test]
+    fn map_of_map() {
+        let store = Store::new(MapStore::new());
+        let mut map: Map<u32, Map<u32, u32>> = Map::create(store.clone(), ()).unwrap();
+
+        map.entry(42).unwrap().or_insert(()).unwrap();
+
+        let mut sub_map = map.get_mut(42).unwrap().unwrap();
+        sub_map.entry(13).unwrap().or_insert(26).unwrap();
+
+        let inner_map = map.get(42).unwrap().unwrap();
+        let actual = inner_map.get(13).unwrap().unwrap();
+
+        assert_eq!(*actual, 26);
+    }
+
+    #[test]
+    fn map_of_map_from_store() {
+        let store = Store::new(MapStore::new());
+        let mut edit_map: Map<u32, Map<u32, u32>> = Map::create(store.clone(), ()).unwrap();
+
+        edit_map.entry(42).unwrap().or_insert(()).unwrap();
+
+        let mut sub_map = edit_map.get_mut(42).unwrap().unwrap();
+        sub_map.entry(13).unwrap().or_insert(26).unwrap();
+
+        edit_map.flush().unwrap();
+
+        let read_map: Map<u32, Map<u32, u32>> = Map::create(store.clone(), ()).unwrap();
+        let inner_map = read_map.get(42).unwrap().unwrap();
+        let actual = inner_map.get(13).unwrap().unwrap();
+
+        assert_eq!(*actual, 26);
+    }
+
+    #[test]
+    fn map_of_deque() {
+        let store = Store::new(MapStore::new());
+        let mut edit_map: Map<u32, Deque<u32>> = Map::create(store.clone(), ()).unwrap();
+
+        edit_map
+            .entry(42)
+            .unwrap()
+            .or_insert(Meta::default())
+            .unwrap();
+
+        let mut deque = edit_map.get_mut(42).unwrap().unwrap();
+        deque.push_front(84).unwrap();
+
+        edit_map.flush().unwrap();
+
+        let mut read_map: Map<u32, Deque<u32>> = Map::create(store.clone(), ()).unwrap();
+        let actual = read_map
+            .get_mut(42)
+            .unwrap()
+            .unwrap()
+            .pop_front()
+            .unwrap()
+            .unwrap();
+
+        assert_eq!(*actual, 84);
```

Comment:
Maybe I missed it, but I don't think there's a test which covers updating a value which is already in the store, then trying to iterate over it (which would have caught an issue in `iter_merge_next`).